### ISSUE TITLE
Sidecar + weave unable to assign IP addresses + fix

### ISSUE THREAD

**Issue by jimpick** [ipfs/testground#667]
#### Version information:

I encountered a problem running my fork of testground that I'm using for Filecoin/Lotus testing: 

https://github.com/jimpick/testground/tree/scratch/lotus-base

It is based on a fork slight after v0.1 came out:

commit 9aa4221a9c3a1859f20e59b308eef0fffce4a246
Author: Anton Evangelatov <anton.evangelatov@gmail.com>
Date:   Mon Feb 17 15:15:51 2020 +0100

    aws ecr repo must be unique across regions (#536)

#### Description:

After rebooting some nodes, some tests which were working previously on my k8s cluster started to fail - sidecar wouldn't get to the 'ip-changed' state:

https://github.com/jimpick/testground/blob/2ea83bfae939a68d66d76d95f63aa43288ea6bf2/plans/basic-tcp/main.go#L113

The sidecar logs showed this error:

```
sidecar worker failed: container worker failed: failed to update network default: failed to add network through cni plugin: unable to allocate IP address: json: cannot unmarshal object into Go struct field ipamConf.ipam.ips of type string	{"host": "unix:///var/run/docker.sock", "container": "5be69f18878ae6fb05850b07a53ee16462bfc44dd8aa1470a9d06c4f03045deb"}
```

#### Fix

After a full day of tracing down the problem, I rebuilt the "weave-kube" docker image from @nonsense 's branch here:

https://github.com/nonsense/weave/tree/support-for-manual-ip

I replaced the `nonsens3/weave-kube` with `jimpick/weave-kube:jim5` here:

https://github.com/ipfs/testground/blob/89e6a268f1263eb9706d26b24ecaff4866611ab8/infra/k8s/kops-weave/weave.yml#L193

That fixes it. My image has some extra logging which helped me to trace down the problem. I'm not really sure what happened, but I suspect the last published `nonsens3/weave-kube` image has some different code that doesn't match the current sidecar code. Things were working for me until I rebooted the node and it pulled the newer image.

After running `kubectl delete -f ./sidecar.yaml -f ./kops-weave/s3bucket.yml -f ./kops-weave/weave.yml -f ./kops-weave/genie-plugin.yaml` ... it's necessary to log into the node via ssh and do `rm -f /opt/cni/bin/weave-*` before reapplying the yaml files. Alternatively, one could probably scale the kops cluster node count to zero, and then scale it back up after reapplying the modified yaml file.

**Comment by raulk** [comment]
This problem could be caused by https://github.com/ipfs/testground/pull/575; I believe the fork you were testing with did not contain that fix. Would be phenomenal if you could try to reproduce with master.

**Comment by raulk** [comment]
@jimpick -- could you describe the fix you applied in your fork of the @nonsense's weave fork, that made this issue go away?

**Comment by jimpick** [comment]
@raulk I just ran `make` in @nonsense's weave fork, and `docker tag weaveworks/weave-kub <my-dockerhub-user>/weave-kube:<tag>` so I could upload it.

**Comment by nonsense** [comment]
@jimpick is right - `nonsens3/weave-kube:latest` was broken. I am not sure what I did wrong, because I built this image weeks ago, and we don't have a process to maintain our weave fork right now.

This would have been caught if we have an integration test for the `network/ping-pong` test, but so far we don't have one.

Also currently it seems like the `network/ping-pong` test is one of the very few that actually uses the `ip-changed` functionality, so this was not caught while running the `dht` testplans.

---

I updated the `nonsens3/weave-kube:latest` image by rebuilding the `support-for-manual-ip` branch at https://github.com/weaveworks/weave and `network/ping-pong` is now working again.

Next steps is to fix:
- https://github.com/ipfs/testground/issues/378
- https://github.com/ipfs/testground/issues/669
